clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -667,6 +683,35 @@ private static <K, V> ClusterResourceListeners configureClusterResourceListeners
         return clusterResourceListeners;
     }
 
+    /**
+     *
+     * Indicates if the consumer is using the Kafka-based offset management strategy,
+     * according to config {@link CommonClientConfigs#GROUP_ID_CONFIG}
+     */
+    private boolean isCommittedOffsetsManagementEnabled() {
+        return groupId.isPresent();
+    }
+
+    /**
+     * Refresh the committed offsets for provided partitions.
+     *
+     * @param timer Timer bounding how long this method can block
+     * @return true iff the operation completed within the timeout
+     */
+    private boolean refreshCommittedOffsetsIfNeeded(Timer timer) {
+        final Set<TopicPartition> initializingPartitions = subscriptions.initializingPartitions();
+
+        log.debug("Refreshing committed offsets for partitions {}", initializingPartitions);
+        try {
+            final Map<TopicPartition, OffsetAndMetadata> offsets = eventHandler.addAndGet(new OffsetFetchApplicationEvent(initializingPartitions), timer);
```

Comment:
We block the `poll` call here to initialize the offset, but when the offset is reset, we handle the offsetReset asynchronously without blocking the `poll` call. Should we do these consistently? Does the existing consumer have the same behavior?

Reply:
You described it correctly, but yes, that's how the existing consumer does it so we wanted to ensure the same behaviour. When initializing positions:
- if using committed offsets, the current consumer blocks on [fetchCommittedOffsets](https://github.com/apache/kafka/blob/f46db86d34f9e5fe1b0d7604306a5108a89c113e/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L996C51-L996C72) until it gets the committed offsets from the coordinator (or the timeout expires). 
- if not using committed offsets, it will just transition the partitions from init to awaiting_reset [here](https://github.com/apache/kafka/blob/f46db86d34f9e5fe1b0d7604306a5108a89c113e/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L2450), and then the positions will be updated following the reset logic, that does not block.